assets/js/components/UserMenu.js

```diff
@@ -0,0 +1,176 @@
+	const userEmail = useSelect( ( select ) => select( CORE_USER ).getEmail() );
+	const userPicture = useSelect( ( select ) => select( CORE_USER ).getPicture() );
+	const splashScreenURL = useSelect( ( select ) => select( CORE_SITE ).getAdminURL( 'googlesitekit-splash', { googlesitekit_context: 'revoked' } ) );
+
+	const [ dialogActive, toggleDialog ] = useState( false );
+	const [ menuOpen, toggleMenu ] = useState( false );
+	const menuButtonRef = useRef();
+	const menuRef = useRef();
+
+	useEffect( () => {
+		global.addEventListener( 'mouseup', handleMenuClose );
+		global.addEventListener( 'keyup', handleMenuClose );
+		global.addEventListener( 'keyup', handleDialogClose );
+
+		return () => {
+			global.removeEventListener( 'mouseup', handleMenuClose );
+			global.removeEventListener( 'keyup', handleMenuClose );
+			global.removeEventListener( 'keyup', handleDialogClose );
+		};
+	}, [] );
+
+	const handleMenu = useCallback( () => {
+		toggleMenu( ! menuOpen );
+	}, [ menuOpen ] );
+
+	const handleMenuClose = useCallback( ( e ) => {
+		if (
+			( ( 'keyup' === e.type && 27 === e.keyCode ) || 'mouseup' === e.type ) &&
+			! menuButtonRef.current.buttonRef.current.contains( e.target ) &&
+			! menuRef.current.menuRef.current.contains( e.target )
+		) {
+			toggleMenu( false );
+		}
+	}, [] );
+
+	const handleDialog = useCallback( () => {
+		toggleDialog( ! dialogActive );
+		toggleMenu( false );
+	}, [ dialogActive ] );
+
+	const handleMenuItemSelect = useCallback( ( index, e ) => {
+		if (
+			( 'keydown' === e.type && ( 13 === e.keyCode || 32 === e.keyCode ) ) || // Enter or Space is pressed.
+			'click' === e.type // Mouse is clicked
+		) {
+			switch ( index ) {
+				case 0:
+					handleDialog();
+					break;
+				case 1:
+					if ( proxyPermissionsURL ) {
+						global.location.assign( proxyPermissionsURL );
+					}
+					break;
+				default:
+					handleMenu();
+			}
+		}
+	}, [ proxyPermissionsURL, handleMenu, handleDialog ] );
+
+	const handleDialogClose = useCallback( ( e ) => {
+		if ( 27 === e.keyCode ) {
+			toggleDialog( false );
+			toggleMenu( false );
+		}
+	}, [] );
+
+	// Log the user out if they confirm the dialog.
+	const handleUnlinkConfirm = useCallback( () => {
+		// Close the modal.
+		toggleDialog( false );
+
+		// Clear caches.
+		clearWebStorage();
+
+		// Navigate back to the splash screen to reconnect.
+		document.location = splashScreenURL;
+	}, [] );
+
+	if ( ! userEmail ) {
+		return null;
+	}
+
+	return (
+		<Fragment>
+			<div className="googlesitekit-dropdown-menu mdc-menu-surface--anchor">
+				<Button
+					ref={ menuButtonRef }
+					className="googlesitekit-header__dropdown mdc-button--dropdown"
+					text
+					onClick={ handleMenu }
+					icon={ userPicture
+						? <i className="mdc-button__icon" aria-hidden="true"><img className="mdc-button__icon--image"
+							src={ userPicture }
+							alt={ __( 'User Avatar', 'google-site-kit' ) } /></i>
+						: undefined
+					}
```

Comment:
Do these need to be defined all on one line? If not let's format it more like we do elsewhere
```js
icon={
	!! userPicture && (
		<i className="mdc-button__icon" aria-hidden="true">
			<img
				className="mdc-button__icon--image"
				src={ userPicture }
				alt={ __( 'User Avatar', 'google-site-kit' ) }
			/>
		</i>
	)
}
```